File: DSA Practice/LeetCode/118. Pascal's Triangle/solution.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        output = [[1]]
        for i in range(1,numRows):
            curr = [1]
            j = 1
            while(j < i):
                curr.append(prev[j-1] + prev[j])
                j+=1
            curr.append(1)
            output.append(curr)
            prev = curr
        return output
            

File: DSA Practice/LeetCode/118. Pascal's Triangle/solution.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
        IList<IList<int>> output = new List<IList<int>>();
        if (numRows == 0)
        {
            return output;
        }

        output.Add(new List<int> { 1 });

        for (int i = 1; i < numRows; i++)
        {
            IList<int> curr = new List<int> { 1 };
            int j = 1;
            IList<int> prev = output[i - 1] as List<int>;

            while (j < i)
            {
                curr.Add(prev[j - 1] + prev[j]);
                j++;
            }

            curr.Add(1);
            output.Add(curr);
        }

        return output;
    }
}